cityofaustin/joplin
# 4165 - tests with fixtures

https://github.com/cityofaustin/techstack/issues/4165

# Description
Adds tests for our v3 service_page factories.

It also sets a cool pattern for handling test fixtures. I'd love to hear what ya'll think and what improvements we could make on this.

Fixtures for content types can be loaded into:
- in tests
- in your local environment
- as "dummy" data in a deployed environment
  - Using Factory fixtures as LOAD_DATA=dummy data will save us from having to make manual django datadump + migration patches for dummy fixtures.

# Types of changes



# How can this be tested?



 

- You can load the dummy data locally with `pipenv run python joplin/manage.py load_test_service_pages`
- Check it out deployed: https://joplin-pr-4165-service-page-te.herokuapp.com/admin/pages/search/

# Checklist:
